addons/mrp/models/mrp_production.py

```diff
@@ -674,10 +678,66 @@ def _get_move_raw_values(self, product_id, product_uom_qty, product_uom, operati
             'propagate_cancel': self.propagate_cancel,
             'propagate_date': self.propagate_date,
             'propagate_date_minimum_delta': self.propagate_date_minimum_delta,
-            'delay_alert': self.delay_alert
+            'delay_alert': self.delay_alert,
+            'workorder_id': workorder_id,
         }
         return data
 
+    def _update_quantity(self, new_qty):
+        for production in self:
+            move_finished_ids = production.move_finished_ids.filtered(lambda m: m.product_id == production.product_id)
+            ongoing_move_raw_ids = production.move_raw_ids.filtered(lambda m: m.state not in ('done', 'cancel'))
+            produced = sum(move_finished_ids.mapped('quantity_done'))
+            if produced > new_qty:
+                precision_digits = - int(log10(move_finished_ids.product_uom.rounding))
+                produced = float_repr(produced, precision_digits=precision_digits)
```

Comment:
what

Reply:
raise UserError(10/3) returns 3.3333333333333335 So when displaying a float in a userError float_repr should be used in order to have a cleaner result